modules/services/repositories/src/main/java/au/com/shiftyjelly/pocketcasts/repositories/subscription/SubscriptionManagerImpl.kt

```diff
@@ -336,14 +336,27 @@ class SubscriptionManagerImpl @Inject constructor(
         freeTrialEligible = eligible
     }
 
-    override fun getDefaultSubscription(subscriptions: List<Subscription>): Subscription? {
-        val trialsIfPresent = subscriptions
+    override fun getDefaultSubscription(
+        subscriptions: List<Subscription>,
+        tier: Subscription.SubscriptionTier?,
+        frequency: SubscriptionFrequency?,
+    ): Subscription? {
+        val subscriptionTier = tier ?: Subscription.SubscriptionTier.PLUS
+        val subscriptionFrequency = frequency ?: SubscriptionFrequency.YEARLY
+
+        val tierSubscriptions = subscriptions.filter { it.tier == subscriptionTier }
+        val trialsIfPresent = tierSubscriptions
             .filterIsInstance<Subscription.WithTrial>()
-            .ifEmpty { subscriptions }
 
         return trialsIfPresent.find {
-            it.recurringPricingPhase is SubscriptionPricingPhase.Months
-        } ?: trialsIfPresent.firstOrNull() // if no monthly subscriptions, just display the first
+            it.recurringPricingPhase is SubscriptionPricingPhase.Months // trial is available for monthly only
```

Comment:
Currently, the trial pricing is shown only for the monthly plan. Trial for the yearly plan will be supported in a future PR.